cip-XXXX-Synchronizer-Migration-to-Canton-3-3/cip-XXXX-Synchronizer-Migration-to-Canton-3-3.md

```diff
@@ -0,0 +1,99 @@
+
+  ## CIP:  CIP-00XX
+<pre>
+  Title: Synchronizer Migration with Downtime to Canton 3.3
+  Author:
+    Curtis Hrischuk
+    Marcin Ziolek
+    Soren Bleikertz
+    Wayne Collier
+  License: CC-1.0
+  Status: Draft
+  Type: Standards
+  Created: 2025-05-14
+</pre>
+
+## Abstract
+
+Splice 0.4.0 introduces new features that rely on upgrading from Canton 3.2 to Canton 3.3. This upgrade require a Canton protocol version upgrade, which will require a Major Upgrade with Downtime action by the Super Validators and Validators.
+
+## Copyright
+
+This CIP is licensed under CC0-1.0: [Creative Commons CC0 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/)
+
+## Specification
+
+Super Validators will upgrade from one Canton release to the other, e.g., 3.2 to 3.3 requiring a special upgrade procedure described under [Synchronizer Upgrades with Downtime](https://docs.dev.sync.global/sv_operator/sv_major_upgrade.html#sv-upgrades). This process requires a synchronized operational procedure involving all Super Validator nodes, and pauses the synchronizer until the upgrade is complete. While paused, no transactions will be sequenced on the synchronizer, and thereby also no activity records are created or rewards minted. This process has been tested extensively and has been completed twice previously on DevNet, TestNet and MainNet, for Canton 3.1 and Canton 3.2.  The procedure will be tested for Canton 3.3 on DevNet and TestNet before it will be carried out on MainNet. The procedure typically completes within three hours.
+
+Validators undergo a similar but simpler procedure to upgrade their nodes described in the [validator documentation](https://docs.dev.sync.global/validator_operator/validator_major_upgrades.html). Validators do not have to do this synchronized with SVs, but until they upgrade, their node will be unable to create new activity records or mint rewards.
+
+The proposed dates for the upgrade are:
```

Comment:
I don't think we usually specify exact dates in CIPs. Imho CIPs describe the type of change, timing of that is then up to the operators to decide.

Reply:
I followed the same content structure as [CIP-0031](https://docs.google.com/document/d/1XpruhfcjdtyZqCKePQLxsYpXju0emim5XxfsxxaaUbo/edit?tab=t.0#heading=h.accy2bb4eod1) which included the dates as well as the instructions.  Amanda, Wayne, and Simon reviewed it without comment.  Could I suggest that we proceed with this CIP and raise it as a question for other CIPs?

Please resolve if you are ok with that course of action.  Otherwise please suggest how you would like to proceed.

Reply:
CIP 31 never made it past the draft phase so I don't think that's a good reference.

If you look at all other recent CIPs, e.g. SV weight changes, token standard, featured app markers, … all of them in the end do require a vote with dates. None of the CIPs specify the dates.

Consider the scenario where the upgrade fails (which happened for the upgrade to 3.2). Do we need to update the CIP and get all SV rights owners to vote on it again? imho no, this is an operation concern handled by the SV operators. But then it should not be voted on as part of the CIP.

Reply:
After discussion, the resolution is to leave it in because the dates provide helpful context for the SVs.